ml-conversational-analytic-tool/featureVector.py

```diff
@@ -143,78 +137,24 @@ def formFeatures(self, export_filename="", export=True):
-            else:
-                export_df.to_csv("exports/" + export_filename, index=False)
-
-        return export_df
-
-    def pullStringConversation(self, export_filename="", export=True):
-        """
-        Function to form conversation string given communication on a pull request
-        Inputs: Name of file to export (string), export flag (boolean)
-        """
-
-        # Store each interaction and pull URL for export
-        string_conversations = []
-        pull_urls = []
-        pull_numbers = []
-
-        for index, row in self.raw_data.iterrows():
-            # Make pull message
-            conversation = "{} ({}) : {}\n{}".format(row["User"], row["Created_At"], row["Title"], row["Body"])
-            temp_df_comments = pd.DataFrame(row['Comments'])
-            temp_df_review_comments = pd.DataFrame(row["Review_Comments"])
-
-            if len(temp_df_comments) > 0 or len(temp_df_review_comments) > 0:
-                all_comments = temp_df_comments.append(temp_df_review_comments)
-                all_comments['Created_At'] = pd.to_datetime(all_comments['Created_At'])
-                all_comments = all_comments.sort_values(by=['Created_At'])  # Sort data so as to export in order
-
-                for comment_index, comment_row in all_comments.iterrows():
-                    conversation = conversation + "\n{} ({}) : {}".format(comment_row["User"],
-                                                                          comment_row["Created_At"],
-                                                                          comment_row["Body"])
-            string_conversations.append(conversation.encode("ascii", "ignore").decode())
-            pull_urls.append(row["URL"])
-            pull_numbers.append(row["Number"])
-
-        # Export converation field dataset
-        export_df = pd.DataFrame()
-        export_df["Number"] = pull_numbers
-        export_df["URL"] = pull_urls
-        export_df["Thread"] = string_conversations
-
-        if export:
-            if not os.path.exists('exports'):
-                os.mkdir('exports')
-            if export_filename == "":
-                export_df.to_csv("exports/" + self.raw_filename + "_analysis.csv", index=False)
-            else:
-                export_df.to_csv("exports/" + export_filename, index=False)
         return export_df
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description='Form features from raw data.')
     parser.add_argument('rawdatafile', help='Raw Data Filename')
-    parser.add_argument('words', help='File containing words to count')
-    parser.add_argument('-unannotated', action='store_true', default=False, help='Create file for annotation')
+    parser.add_argument('-w', '--words', required=False, help='File containing words to count')
+    parser.add_argument('-n', '--name', required=False, help='Output file name. If not specified, a default '
```

Comment:
You can set a default for argparse by using the "default" kwarg

Reply:
You can set default=rawdatafile + ".csv"

Reply:
By settting `default=rawdatafile + ".csv"` we will attempt to overwrite the source rawdatafile. 
- We could set some default name to be used in case the name is not explicitly specified - this means that whatever the source data file is, the output file name will be always the same.
- We could stick with the current implementation but shorten the help message to   `Output file name.`
Provided or not, the output file will be printed out.  
If not provided the filename is constructed this way -`<rawdatafile>_analysis.csv`

Reply:
I'm simply suggesting setting the default="" to whatever value is used as the default if the name is not provided. Currently, it is a mystery to the user what filename is the default.  By adding it to argparse, you can print it out to the CLI help.

Reply:
Done. 
In addition, I've refactored `utils.export_to_cvs` - extracted output file name generation in a separate func